gap/callwithtimeout.gd

```diff
@@ -0,0 +1,54 @@
+#############################################################################
+##
+#F  IO_CallWithTimeout( <timeout>, <func>[, <arg1>[, <arg2>....]] )  
+##         . . call a function with a time limit
+#F  IO_CallWithTimeoutList( <timeout>, <func>, <arglist> )  
+##
+##  <#GAPDoc Label="IO_CallWithTimeout">
+##  <Index>Timeouts</Index>
+##  <ManSection>
+##  <Func Name="IO_CallWithTimeout" Arg='timeout, func, .....'/>
+##  <Func Name="IO_CallWithTimeoutList" Arg='timeout, func, arglist'/>
+##
+##  <Description>
+##    <C>IO_CallWithTimeout</C> and <C>IO_CallWithTimeoutList</C> support calling a function
+##  with a limit on length of time it will run. The function is run inside a copy of the current GAP
+##  session, so any changes it makes to global variables are thrown away when the function finishes
+##  or times out.<P/>
+##
+##  <C>IO_CallWithTimeout</C> is variadic. 
+##  Its third and subsequent arguments, if any, are the arguments passed to <A>func</A>.
+##  <C>IO_CallWithTimeoutList</C> in contrast takes exactly three arguments, of which the third is a list
+##  (possibly empty) or arguments to pass to <A>func</A>. <P/>
+##
+##  If the call completes within the allotted time and returns a value <C>res</C>, the result of 
+##  <C>IO_CallWithTimeout[List]</C> is a length 2 list of the form <C> [ true, res ] </C>. <P/>
+##  
+##  If the call completes within the allotted time and returns no value, the result of 
+##  <C>IO_CallWithTimeout[List]</C> is a list of length 1 containing the value <C>true</C>.<P/>
+##
+##  The call occurs with 
+##  If the call does not complete within the timeout, the result of <C>IO_CallWithTimeout[List]</C>
+##  is a list of length 1 containing the value <C>false</C>  In this case, just as if you had
+##  <C>quit</C> from a break loop, there is some risk that internal data structures in &GAP; may
+##  have been left in an inconsistent state, and you should proceed with caution.<P/>
```

Comment:
Wait, I thought the computation runs in a forked process, so why is this an issue?